example/node_classification.py

```diff
@@ -0,0 +1,131 @@
+import torch
+from torch_geometric.nn import GCNConv
+import torch.nn.functional as F
+from torch_geometric.datasets import Planetoid
+from torch_geometric.transforms import NormalizeFeatures
+import mgp
+import numpy as np
+from tqdm import tqdm
+
+class Data(object):
+    def __init__(self, x, y, edge_index, train_mask, val_mask, test_mask):
+        self.x = x
+        self.y = y
+        self.edge_index = edge_index
+        self.train_mask = train_mask
+        self.val_mask = val_mask
+        self.test_mask = test_mask
+
+class GCN(torch.nn.Module):
+    def __init__(self, hidden_channels):
+        super().__init__()
+        torch.manual_seed(1234567)
+        self.conv1 = GCNConv(1433, hidden_channels)
+        self.conv2 = GCNConv(hidden_channels, 7)
+        
+
+    def forward(self, x, edge_index):
+        x = self.conv1(x, edge_index)
+        x = x.relu()
+        x = F.dropout(x, p=0.5, training=self.training)
+        x = self.conv2(x, edge_index)
+        return x
+
+
+
+
+
+def train(data):
+    model.train()
+    optimizer.zero_grad()  # Clear gradients.
+    out = model(data.x, data.edge_index)  # Perform a single forward pass.
+    loss = criterion(out[data.train_mask], data.y[data.train_mask])  # Compute the loss solely based on the training nodes.
+    loss.backward()  # Derive gradients.
+    optimizer.step()  # Update parameters based on gradients.
+    return loss
+
+
+
+def test(data, mask):
+    model.eval()
+    out = model(data.x, data.edge_index)
+    pred = out.argmax(dim=1)  # Use the class with highest probability.
+    correct = pred[mask] == data.y[mask]  # Check against ground-truth labels.
+    acc = int(correct.sum()) / int(mask.sum())  # Derive ratio of correct predictions.
+    return acc
+
+def convert_data(nodes, edges):
+    if len(nodes) == 0:
+        return "Insert data in database first!"
```

Comment:
Situations like this are better handled by throwing an exception.

Reply:
I was afraid of throwing exceptions, thank you for enlightment. I've repaired now. (in new branch)